handler/category-dimension-import.go

```diff
@@ -183,3 +154,45 @@ func (h *CategoryDimensionImport) instanceFailed(ctx context.Context, err error,
 	}
 	return err
 }
+
+// onLastDimension handles the case where all dimensions have been updated to the instance.
+// If the eTag did not change since the last update, then we know that this consumer was the last one to update the instance. In that case:
+// - Set instance to edition-confirmed
+// - Set import job to completed state
+// - send an InstanceComplete kafka message
+func (h *CategoryDimensionImport) onLastDimension(ctx context.Context, e *event.CategoryDimensionImport, eTag string) error {
+
+	// set instance to 'edition-confirmed' state, only if the eTag value did not change
+	_, err := h.datasets.PutInstanceState(ctx, h.cfg.ServiceAuthToken, e.InstanceID, dataset.StateEditionConfirmed, eTag)
+	if err != nil {
+		switch putErr := err.(type) {
+		case *dataset.ErrInvalidDatasetAPIResponse:
+			if putErr.Code() == http.StatusConflict {
+				log.Info(ctx, "instance state could not be set to 'edition-confirmed' because it changed between putting the last dimension option and counting")
+				return nil // valid scenario, another consumer was the last one to update the post instances.
+			}
+		}
+		return &Error{
+			err:     fmt.Errorf("error while trying to set the instance to edition-confirmed state: %w", err),
+			logData: log.Data{"event": e},
+		}
+	}
+
+	// Import api update job to completed state
+	if err := h.importApi.UpdateImportJobState(ctx, e.JobID, h.cfg.ServiceAuthToken, StateImportCompleted); err != nil {
+		return fmt.Errorf("error updating import job to completed state: %w", err)
+	}
+
+	// create InstanceComplete event and Marshal it
+	bytes, err := schema.InstanceComplete.Marshal(&event.InstanceComplete{
+		InstanceID: e.InstanceID,
+	})
+	if err != nil {
+		return err
```

Comment:
So what happens to clean up this failed situation ?

Reply:
In this case, the job and instance states are updated, but the kafka message is not sent. 
If we observe this case (marshal fails), it means that there is a bug in the schema, so it's not something we should retry.